packages/presentation/lib/app/app_bloc.dart

```diff
@@ -46,7 +54,14 @@ class _AppBloc extends BlocImpl implements AppBloc {
 
   @override
   void onItemTapped(int index) {
-    selectedIndex = index;
+    _selectedIndex = index;
```

Comment:
I think you need to check current index and push new page only if old and new indexes are different, because you will be removing all pages from stack and create one from scratch every time when you don't need it.

Reply:
But if I go to details page(and then forward forward) and then want to get back to homeScreen by tapping the bottom nav? Will this check allow me to do this?

Reply:
Possible we need add additional checking if details page on top of stack then remove one from it if home index and selected index are the same.

Reply:
the worst variant is to Add check if selected index 0 and screen is already homescreen, then do not pop and push it again - it is bad from ux side
better variant is as it is not
the best variant - add additional checking if details page on top of stack then remove one from it if home index and selected index are the same.

Reply:
but it will work only if we will not add new nested screens after details( eg cast list, photos). If so we would need to expand our check

Reply:
details is also nested screen

Reply:
so should be used something like pop until. if some screens will be on top of details